File: OOAD/solid-principle-app/src/ocp/voilation/FixedDeposit.java
package ocp.voilation;

public class FixedDeposit {
	private int accountNumber;
	private String name;
	private double principle;
	private int years;
	FestivalType festival;
	
	public FixedDeposit(int accountNumber,String name,double principle,int years,FestivalType festival){
		this.accountNumber=accountNumber;
		this.name=name;
		this.principle=principle;
		this.years=years;
		this.festival=festival;
	}
	
	public double calculateTotalInterest(){
		if(festival.equals(FestivalType.HOLI)){
			return (principle*years*(8.5))/100;
		}
		if(festival.equals(FestivalType.NEW_YEAR)){
			return (principle*years*(8))/100;
		}
		if(festival.equals(FestivalType.NORMAL)){
			return (principle*years*(7))/100;
		}
		return -1;
	}
	public int getAccountNumber() {
		return accountNumber;
	}
	public void setAccountNumber(int accountNumber) {
		this.accountNumber = accountNumber;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
}

File: C#/OOP/SOLID Principle/OpenClosed-App/OpenClosed-App/FixedDeposit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenClosed_App
{
    class FixedDeposit
    {
        private int accountNumber;
        private String name;
        private double principle;
        private int years;
        IFestival festival;

        public FixedDeposit(int accountNumber, String name, double principle,
                int years, IFestival festival)
        {
            this.accountNumber = accountNumber;
            this.name = name;
            this.principle = principle;
            this.years = years;
            this.festival = festival;
        }

        public double calculateTotalInterest()
        {
            double rate = festival.GetRate();
            return (principle * years * rate) / 100;
        }

        public int AccountNumber
        {
            get
            {
                return accountNumber;
            }
            set { accountNumber = value; }
        }

        public string Name {
            get { return name; }
            set { name = value; }
        }

    }
}
